### PR TITLE
Integration tests for devenv-devcontainer-setup

### PR DESCRIPTION
**Pull request by dennismeister93**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -70,3 +70,88 @@ jobs:
         if: always()
         with:
           junit_files: ./results/UnitTest/junit.xml
+
+  run-integration-tests:
+    name: Run Integration Test
+    runs-on: ubuntu-latest
```
Comment (MP91) [inline .github/workflows/ci.yml:76]: please don't use latest, fix the version

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -70,3 +70,88 @@ jobs:
...
+    container: ghcr.io/eclipse-velocitas/devcontainer-base-images/${{ matrix.language }}:latest
+    strategy:
+      matrix:
+        language: ["python", "cpp"]
+      fail-fast: false
```
Comment (MP91) [inline .github/workflows/ci.yml:81]: why not failing fast?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -70,3 +70,88 @@ jobs:
...
+      - name: Checkout template repo
+        uses: actions/checkout@v3
+        with:
+          repository: eclipse-velocitas/vehicle-app-${{ matrix.language }}-template
+          path: test/${{ matrix.language }}/repo
+
+      - name: Prepare .velocitas.json
+        run: |
+          NEW_CONFIG="$(jq --arg GITHUB_SHA "$GITHUB_SHA" '.packages[0].version |= $GITHUB_SHA' test/${{ matrix.language }}/.velocitas.json)"
+          echo "${NEW_CONFIG}" > test/${{ matrix.language }}/.velocitas.json
```
Comment (MP91) [inline .github/workflows/ci.yml:94]: why don't you write directly to the repo?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -70,3 +70,88 @@ jobs:
...
+
+      - name: Sync velocitas project
+        run: |
+          cd test/${{ matrix.language }}/repo
+          velocitas sync
+          if [ $? -ne 0 ];
+          then
+            exit 1
+          fi
+          git checkout .velocitas.json
```
Comment (MP91) [inline .github/workflows/ci.yml:122]: please add a comment what is happening here

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -70,3 +70,88 @@ jobs:
...
+          git diff
+          if [[ -z "$(git status --porcelain .)" ]];
+          then
+            echo "No Changes"
+            echo "changed=0" >> $GITHUB_OUTPUT
+          else
+            echo "Changes"
+            echo "changed=1" >> $GITHUB_OUTPUT
+          fi
+        shell: bash
```
Comment (MP91) [inline .github/workflows/ci.yml:151]: why do we need these both steps? couldn't we achieve the same with only one of them?

**Review by MP91 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dennismeister93) [reply .github/workflows/ci.yml:81]: to identify and debug it better in case of an error which is "only" in one of the matrix

**Review by MP91 — APPROVED** [review]
LGTM

**Pull request merged**